rest_framework/request.py

```diff
@@ -86,7 +86,7 @@ def clone_request(request, method):
     Internal helper method to clone a request, replacing with a different
     HTTP method.  Used for checking permissions against other methods.
     """
-    ret = Request(request=request._request,
+    ret = Request(request=request,
```

Comment:
Good start here.

Rather than simply pass through the entire request, we should instead make sure to just set a `version` attribute on the cloned request, where needed. Having a `Request` object with a `._request=Request` which in turn has a `._request=HttpRequest` would be a bit weird.